jaxtronomy/ImSim/image_linear_solve.py

```diff
@@ -0,0 +1,386 @@
+        linear_prior=None,
+        check_positive_flux=False,
+        linear_solver=False,
+    ):
+        """Computes the likelihood of the data given a model This is specified with the
+        non-linear parameters and a linear inversion and prior marginalisation.
+
+        :param kwargs_lens: list of keyword arguments corresponding to the superposition
+            of different lens profiles
+        :param kwargs_source: list of keyword arguments corresponding to the
+            superposition of different source light profiles
+        :param kwargs_lens_light: list of keyword arguments corresponding to different
+            lens light surface brightness profiles
+        :param kwargs_ps: keyword arguments corresponding to "other" parameters, such as
+            external shear and point source image positions
+        :param source_marg: bool, performs a marginalization over the linear parameters
+        :param linear_prior: linear prior width in eigenvalues
+        :param check_positive_flux: bool, if True, checks whether the linear inversion
+            resulted in non-negative flux components and applies a punishment in the
+            likelihood if so.
+        :param linear_solver: bool, if True (default) fixes the linear amplitude
+            parameters 'amp' (avoid sampling) such that they get overwritten by the
+            linear solver solution.
+        :return: log likelihood (natural logarithm), linear parameter list
+        """
+        # generate image
+        if linear_solver is False:
+            im_sim = super(ImageLinearFit, self).image(
+                kwargs_lens,
+                kwargs_source,
+                kwargs_lens_light,
+                kwargs_ps,
+                kwargs_extinction,
+                kwargs_special,
+            )
+            cov_matrix, param = None, None
+            model_error = self._error_map_model(
+                kwargs_lens, kwargs_ps=kwargs_ps, kwargs_special=kwargs_special
+            )
+        else:
+            raise ValueError(
+                "Linear solver not supported in JAXtronomy. Use lenstronomy instead"
+            )
+        # compute X^2
+        logL = self.likelihood_data_given_model_solution(
+            im_sim,
+            model_error,
+            cov_matrix,
+            param,
+            kwargs_lens,
+            kwargs_source,
+            kwargs_lens_light,
+            kwargs_ps,
+            source_marg=source_marg,
+            linear_prior=linear_prior,
+            check_positive_flux=check_positive_flux,
+        )
+        return logL, param
+
+    def likelihood_data_given_model_solution(
+        self,
+        model,
+        model_error,
+        cov_matrix,
+        param,
+        kwargs_lens,
+        kwargs_source,
+        kwargs_lens_light,
+        kwargs_ps,
+        source_marg=False,
+        linear_prior=None,
+        check_positive_flux=False,
+    ):
+        """
+
+        :param model:
```

Comment:
I guess these docstrings were not there in lenstronomy itself. Would it be possible to in parallel adding them to lenstronomy as well? Thank you!